utils/http.go

```diff
@@ -112,6 +176,20 @@ func buildAccessRecords(repo string, actions ...string) []auth.Access {
 	return requiredAccess
 }
 
+func buildCatalogRecord(actions ...string) []auth.Access {
+	requiredAccess := make([]auth.Access, 0, len(actions))
+	for _, action := range actions {
+		requiredAccess = append(requiredAccess, auth.Access{
+			Resource: auth.Resource{
+				Type: "registry",
+				Name: "catalog",
```

Comment:
The auth lib and the token servers know how to handle this. According to @jlhawn this is the correct resource to request for listing all repos.